src/anyp/Uri.h

```diff
@@ -54,7 +54,7 @@ class Uri
         hostIsNumeric_ = false;
         *host_ = 0;
         hostAddr_.setEmpty();
-        port_ = 0;
+        port_ = std::nullopt;
```

Comment:
```suggestion
        port_.reset();
```

Reply:
Why do you think reset() is better here?

PR code feels slightly clearer to me -- it explicitly sets `port_` to a well-known "no value" state. The proposed change will result in the same "no value" state, of course, but it is slightly less clear IMO because the reader may not remember the `port_` type and might suspect that its reset() method changes the port to some known default value like 80 or 0.

Reply:
There is no big difference, as you noted, but it seems to me that reset() is more in line with the calling clear() method - i.e., resetting all values to the 'default' state (equivalent to the default-constructed object). 

> a well-known "no value" state

Methods such as 'reset()' are common for STL types (e.g., smart pointers also have them). On the other hand,  'default' types (such as nullopt) are specific for each type.

Reply:
> Methods such as 'reset()' are common for STL types... On the other hand, 'default' types (such as nullopt) are specific for each type.

I agree that using the same (common) method for many data members is an advantage. IMO, being clear that there is no default port here outweighs that advantage in this std::optional context. We seem agree that the difference is small and unimportant, so if others prefer reset() here, I will accept your suggestion instead.